servers/express.js

```diff
@@ -63,22 +63,22 @@ apiRoutes.post('/logfiles', upload.any(), async(req, res) => {
   for (const file of req.files) {
     const filepath = path.join(__dirname, '..', file.path);
     try {
-      const contents = await fsp.readFile(filepath, 'utf-8');
-      const firstRecord = JSON.parse(contents.slice(0, contents.indexOf('\n')));
-      const headerProps = ['ts', 'type', 'entry'];
-      const recordProps = Object.getOwnPropertyNames(firstRecord);
-
-      if (!headerProps.every(e => recordProps.includes(e))) {
-        throw new Error('Invalid File');
-      } else if (firstRecord.type != 'header') {
-        throw new Error('Invalid File');
+      var contents = await fsp.readFile(filepath, 'utf-8');
+      var firstRecord = JSON.parse(contents.slice(0, contents.indexOf('\n')));
+      var headerProps = ['ts', 'type', 'entry'];
+      var recordProps = Object.getOwnPropertyNames(firstRecord);
+
+      if (!headerProps.every(e => recordProps.includes(e)) && firstRecord.type != 'header') {
+        throw new Error('Invalid logile');
       }
+      file.status = {uploaded: true, reason: ''};
     } catch (err) {
+      file.status = {uploaded: false, reason: 'Invalid logfile!'};
```

Comment:
probably want to use `err` as opposed to set a hardcoded value. and it would be better if the message were more explicit - maybe either "missing header", "unknown version", or "malformed header record".